### ISSUE TITLE
Traefik 1.7.31 does not work with preferredChain

### ISSUE THREAD

**Issue by tgerakitis** [traefik/traefik#8494]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

* shut down `traefik` by running `docker-compose down` 
* changes my `traefik.toml` to include 
* deleted the `acme.json` 
* prepared a new, empty `acme.json`
  * `touch acme.json`
  * `chmod 600 acme.json`
* started traefik again `docker-compose up -d`

### What did you see instead?

when checking the certificate on `sslabs.com` or via the following command, I see the X3 path still present
```console
$ openssl s_client -connect my.example.com:https | head
> depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
> verify return:1
> depth=1 C = US, O = Let's Encrypt, CN = R3
> verify return:1
> depth=0 CN = my.example.com
> verify return:1
> CONNECTED(00000003)
> ---
> Certificate chain
>  0 s:CN = my.example.com
>    i:C = US, O = Let's Encrypt, CN = R3
>  1 s:C = US, O = Let's Encrypt, CN = R3
>    i:C = US, O = Internet Security Research Group, CN = ISRG Root X1
>  2 s:C = US, O = Internet Security Research Group, CN = ISRG Root X1
>    i:O = Digital Signature Trust Co., CN = DST Root CA X3
> ---
> 
```

### What version of Traefik are you using?

1.7.31

### What is your environment & configuration?

```yaml
version: '2.4'

services:
  traefik:
    image: traefik:1.7.31
    restart: always
    environment:
      GODEBUG: tls13=1
    networks:
      - web
      - default
    ports:
      - 80:80
      - 443:443
    labels:
      traefik.enable: false
    volumes:
      - "./traefik.toml:/traefik.toml"
      - "./acme.json:/acme.json"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

`traefik.toml`:
```toml
defaultEntryPoints = ["http","https"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
   [entryPoints.https.tls]
   minVersion = "VersionTLS12"
   cipherSuites = [
       #TLS 1.3
       "TLS_AES_128_GCM_SHA256",
       "TLS_AES_256_GCM_SHA384",
       "TLS_CHACHA20_POLY1305_SHA256",
       #TLS 1.2
       "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
       "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
       "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
       "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
       "TLS_RSA_WITH_AES_128_GCM_SHA256",
       "TLS_RSA_WITH_AES_256_GCM_SHA384",
       "TLS_RSA_WITH_AES_128_CBC_SHA",
       "TLS_RSA_WITH_AES_256_CBC_SHA"
   ]


[accessLog]

[retry]

[api]

[docker]
  endpoint = "unix:///var/run/docker.sock"
  domain = "example.com"
  watch = true
  exposedbydefault = false
[acme]
  email = "office@example.com"
  storage = "acme.json"
  entryPoint = "https"
  onHostRule = true
  preferredChain = "ISRG Root X1"

[acme.httpChallenge]
  entryPoint = "http"
```

### If applicable, please paste the log output in DEBUG level

Related to  #8480

**Comment by iasmanis** [comment]
Seeing the same result after upgrade and requesting new certificate. Are you supposed to request a new certificate after configuring `preferredChain` or will this work with previously issued certificates as well (we are using consul storage)?

**Comment by traefiker** [comment]
Closed by #8495.
